problem_builder/table.py

```diff
@@ -66,23 +75,148 @@ class MentoringTableBlock(StudioEditableXBlockMixin, StudioContainerXBlockMixin,
             {"display_name": "Immunity Map", "value": "immunity-map"},
         ],
     )
-    editable_fields = ("type", )
+    editable_fields = ("type", "allow_download")
+    allow_download = Boolean(
+        display_name=_("Allow Download"),
+        help=_("Allow students to download a copy of the table for themselves."),
+        default=False,
+        scope=Scope.content
+    )
+    allow_sharing = Boolean(
+        display_name=_("Allow Sharing"),
+        help=_("Allow students to share their results with other students."),
+        default=True,
+        scope=Scope.content
+    )
+    # Stores who a user has allowed to see their table.
+    shared_with = Dict(scope=Scope.user_state_summary)
+    # Stores whose tables a user is allowed to view. This needs to be kept in sync with the previous field.
+    view_options = Dict(scope=Scope.user_state_summary)
```

Comment:
Since we're in a time crunch I would suggest storing this data in a Django table instead. And get rid of the username->anonymous ID map table; you can include that data in the new shares table if necessary.